### DISCUSSION TITLE
any issue replacing pointer with unique pointer in this example?

### DISCUSSION THREAD

**Discussion by ThomasCartier** [uNetworking/uWebSockets#1669] [Q&A] [answered]
hi,

in [this example](https://github.com/uNetworking/uWebSockets/blob/20b3ea468502bc9912972998f7650f6505a6eb92/examples/Http3Server.cpp#L54)

is replacing
`res->onData([res, bodyBuffer = (std::string *)nullptr](std::string_view chunk, bool isLast) mutable {`
with
`res->onData([res, bodyBuffer = unique_ptr<std::string>()](std::string_view chunk, bool isLast) mutable {`

a mistake ?

I guess just using a `std::string`
`res->onData([res, bodyBuffer = std::string()](std::string_view chunk, bool isLast) mutable {`
should not be a mistake because the string is copied on the lambda and should not be lost ?

I just want to make sure because there are a lot of pieces rolling.

Thanks

**Comment by uNetworkingAB** [comment, marked as answer]
The size of a pointer is 8 bytes, same for unique_ptr. The size of std::string is 32 bytes. The size of std::function is 32 bytes.

This means that you'd pay allocation cost in your fast path with your change. The whole point is to have a fast fast path and a slow slow path. Whatever you do (unique_ptr is probably fine), you don't want to incur a cost in your fast path.

**Comment by ThomasCartier** [comment]
Thanks a lot, great lib

**Reply by uNetworkingAB** [reply]
unique_ptr is probably best. And combined with make_unique then you have 1 allocation only

**Reply by ThomasCartier** [reply]
👍🙂 thanks a lot!
